packages/ember-principled-forms/tests/dummy/app/components/form-field/template.hbs

```diff
@@ -1,35 +1,83 @@
 {{#docs-demo as |demo|}}
-  {{#demo.example name='eager-principled-form-field.hbs'}}
-    {{principled-form-field
-        type=this.user.name.type
-        value=this.user.name.value
+  {{#demo.example name='eager-required-principled-form-field.hbs'}}
+    {{#principled-form-field
         label='Name'
-        onChange=(action 'handleChange' 'name')
+        type=this.user.name.type
         onInput=(action 'handleInput' 'name')
+        as |field|
     }}
+      <label>
+        {{field.label}}
+        <input
+          type={{field.type}}
+          oninput={{field.onInput}}
```

Comment:
I note that you didn't bind `onchange` here – oversight or intentional?

Reply:
In the specific case of text inputs, isn't `onchange` covered by `oninput`? I didn't see a need to have both. We may not ever need both.

Reply:
Actually, I just thought of a reason to want both. For an email, if you leave the input with `foo@bar.`, then you want validation to fail.